source/standards/alerting.html.md.erb

```diff
@@ -1,77 +1,83 @@
+Services at GDS should send automated alerts to staff if the service's monitoring detects issues.
 
-Alerts can take many forms including emails, phone calls and status
-websites.
+Sending alerts help the service meet service level agreements and user needs.
 
-The service manual has [some information on writing alerts][service_manual_alerts].
+## When to send an alert
 
-[service_manual_alerts]: https://www.gov.uk/service-manual/technology/monitoring-the-status-of-your-service#writing-alerts
+You should send an alert when your [service monitoring][] detects an issue that:
 
-## User needs for alerting
+* affects service users
+* requires action to fix
+* lasts for a sustained period of time
 
-- Support GDS' goal of services being available to meet user needs
-- Mitigate reputational damage to GDS
-- Adhere to service level agreements (with reliant parties like other government departments), industry standards (like PCI compliance) and legal requirements (like the [Code of Practice for Official Statistics][stats_law])
+You should setup alerts based on things affecting users, for example, if service requests are slow or webpages are unavailable. You should still track other things like disks being full or almost full, or databases being down, as these can lead to more serious alerts that do affect users.
 
-[stats_law]: https://gss.civilservice.gov.uk/about/code-of-practice/
+Only things that need action should trigger an alert. This can be an immediate action for critical or high priority alerts, or “soon” for non-urgent alerts.
 
-## Principles for setting up an alerting system
+If you do not need to take action, it should not be an alert. For example, an “alert” that shows the status of a system is actually a monitoring tool and a dashboard can display this instead.
 
-Alerts should be meaningful and actionable. An alert that shows
-the status of a system is actually a monitoring tool. The text of
-your alerts should be specific - people will be responding to alerts
-in the middle of the night.
+Alerts must show something is wrong for a sustained period of time, instead of a brief issue caused by temporary network conditions. For example, Prometheus supports this using the `for` parameter in its alerting rule, which indicates the condition has to be true for a set period of time before an alert's triggered.
 
-Don't rely on emails as an automated alerting method for
-incident responders. Emails become noisy which over time
-results in them being ignored or overlooked. If you absolutely
-have to use emails, template them so that the information is
-presented consistently.
+Specific examples of issues that should not trigger an alert include an individual container instance dying or a single task invocation failing. The orchestration system will bring the instance back up, and the task will retry. A long period of fewer than expected instances or repeated task failures should trigger alerts, as they show underlying problems.
```

Comment:
'The orchestration system' - does this refer to Prometheus or a general principle?

Reply:
General principle. It's talking about the systems that run the underlying architecture (ECS, Kubernetes etc).